src/main/java/jose/Storage.java

```diff
@@ -0,0 +1,65 @@
+package jose;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import jose.task.*;
+
+/**
+ * A class that handles the reading and writing of data stored in the data file.
+ */
+public class Storage {
+    private File file;
+
+    /**
+     * The main constructor that also creates a data file and its relevant directories if one does not already exist.
+     *
+     * @param filePath The path of the data file.
+     * @throws IOException If there are any errors creating the data file.
+     */
+    public Storage(String filePath) throws IOException {
+        file = new File(filePath);
+        if (!file.exists()) {
+            file.getParentFile().mkdirs();
+            file.createNewFile();
+        }
+    }
+
+    /**
+     * Reads the data file and compiles all the tasks saved in it for further processing.
+     *
+     * @return An ArrayList containing the tasks stored in the data file.
+     * @throws IOException If there are any errors accessing the data file.
+     */
+    public ArrayList<String> load() throws IOException {
+        BufferedReader br = new BufferedReader(new FileReader(file));
+        ArrayList<String> tasks = new ArrayList<>();
+        String s;
+
+        while ((s = br.readLine()) != null) {
```

Comment:
I like the way you section the code block, despite the fact that there are not many lines. The separation helps readers/developers understand that differences between those code blocks. 

I see many of such instances in the rest of the code. 👍